diff --git a/dist/core/utils/content-scripts.mjs b/dist/core/utils/content-scripts.mjs
index 77d9457a4c3d130b56c0d793f98b033a5615f7f6..d9a03cb725e33caa1a263ce7b15ddb2dfa191939 100644
--- a/dist/core/utils/content-scripts.mjs
+++ b/dist/core/utils/content-scripts.mjs
@@ -22,7 +22,8 @@ export function hashContentScriptOptions(options) {
   };
   return JSON.stringify(
     Object.entries(withDefaults).map(([key, value]) => {
-      if (Array.isArray(value)) return [key, value.sort()];
+      // Keeping matches unsorted improves metadata in extension stores
+      if (Array.isArray(value) && key !== "matches") return [key, value.sort()];
       else return [key, value];
     }).sort((l, r) => l[0].localeCompare(r[0]))
   );
